synapse_antispam/mjolnir/antispam.py

```diff
@@ -115,6 +115,11 @@ def user_may_create_room_alias(self, user_id, room_alias):
     def user_may_publish_room(self, user_id, room_id):
         return True  # allowed
 
+    def check_username_for_spam(self, user_id, display_name):
+        # Check whether the user ID or display name matches any of the banned
+        # patterns.
+        return self.is_user_banned(user_id) or self.is_user_banned(display_name)
```

Comment:
We should only be checking user ID here (ban list rules don't specify anything about display names).

Reply:
From my conversation with @lampholder it is fairly important to also filter via display names. Do you have any thought on how hard it would be to add that to ban rules?

Reply:
It wouldn't be easy as this is powered by spec, but it could be done. For now I'd recommend that we do that as a different PR/stream of work though, to avoid blocking this further.

Reply:
Makes sense! Thanks!

Reply:
So, I re-read the MSC last night and it doesn't care as much as I thought it did. We could absolutely check for display names here, and probably should.

Reply:
I believe based on our conversation over Matrix that you believe checking both of these is OK now, so I've left it. Please let me know if I've misunderstood!